### PR TITLE
#16: Added Tags UI

### LINKED ISSUE #16: UI for tag selector
Tag selector is a widget which has collection of tags, knows which images are labeled with which tags, and can tell the Grid view new selection, i.e. collection of selected images:
* When user clicks a tag it becomes activated
* When user clicks a tag again, it deactivates
* Selection is updated each time active tags set changes

UI consists of:
* on/off button
* tags grid

tag-selector

### PR DESCRIPTION
**Pull request by prokawsar**
Implements:
- #16

Screenshots
https://ibb.co/6m8FTMC
https://ibb.co/C5Y7rtm

- [x] Tag list visible when filtered by tag 
- [x] Clicking on tag will show only matched with tag
- [x] Deselect tag will show all images
- [x] If selected tag has no image, will show 'No Image found'
- [x] Adding new tag will show on the list instantly (after 1.5 sec)   
- [x] Tag deletion functionality
- [x] Tag deletion question popup

### CODE REVIEW

**Review by alvinosh — APPROVED** [review]
Looks good to me

File: src/lib/components/QuestionModal.svelte [hunk 1]
```diff
@@ -1,19 +1,36 @@
 <script lang="ts">
 	import { faMultiply } from '@fortawesome/free-solid-svg-icons'
 	import Fa from 'svelte-fa'
-	import { onMount } from 'svelte'
+	// import { onMount } from 'svelte'
```
Comment (kirillt) [inline src/lib/components/QuestionModal.svelte]: Commented-out code.

File: src/lib/components/Image.svelte [hunk 2]
```diff
@@ -1,13 +1,45 @@
...
+			$galleryStore.questionModalProp = 'deleteTag'
+			$galleryStore.modal = true
+			return
+		}
+	}
+
+	const handleKeyUp = () => {
+		clearTimeout(typingTmr)
+		typingTmr = setTimeout(() => {
+			// console.log('saving tag')
```
Comment (kirillt) [inline src/lib/components/Image.svelte]: Commented-out code.

File: src/lib/store/index.ts [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
 export const galleryStore = writable(galleryInterface)
+
+// export const galleryStore = {
+// 	subscribe: store.subscribe,
+// 	set: store.set,
+// 	update: store.update,
+// 	reset: () => {
+// 		store.set(JSON.parse(JSON.stringify(galleryInterface)))
+// 	}
+// }
```
Comment (kirillt) [inline src/lib/store/index.ts]: Commented-out code.

File: src/lib/utils/localstorage.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	}
+
+	// Delete operation
+	// delete(name: string): void {
+	// 	const items = this.getAll()
+	// 	if (items.indexOf(name)) {
+	// 		localStorage.setItem(this.storageKey, JSON.stringify(items))
+	// 	}
+	// }
+}
```
Comment (kirillt) [inline src/lib/utils/localstorage.ts]: Commented-out code.

File: src/routes/+page.svelte [hunk 5]
```diff
@@ -5,12 +5,15 @@
...
+	import type { ImageType } from '$lib/utils/types'
+	import TagsList from '$lib/components/TagsList.svelte'
 
-
-	let images: any = []
+	let images: ImageType[] = []
 
 	const uploadFolder = async () => {
 		images = []
+		// galleryStore.reset()
```
Comment (kirillt) [inline src/routes/+page.svelte]: Commented-out code.

**Review by kirillt — CHANGES_REQUESTED** [review]
Looks nice, but let's not have commented-out code in `main`. Such things pile up and reduce maintainability.

**Review by kirillt — APPROVED** [review]

**Pull request merged**